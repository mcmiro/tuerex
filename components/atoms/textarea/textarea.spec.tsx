/* eslint-disable no-use-before-define */
import * as React from 'react';
import { screen, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Textarea from './';

describe('Test Textarea component', () => {
  it('Should pass when name, variant, size and placeholder is additional', () => {
    render(<Textarea variant="outline" placeholder="test" />);

    const texarea = screen.getByRole('textbox');
    expect(texarea).toBeDefined();
    expect(texarea).toMatchSnapshot();
  });

  it('Should pass when value is empty', () => {
    const changeMock = jest.fn();
    render(
      <Textarea
        onInputChange={changeMock}
        variant="outline"
        placeholder="test-placeholder"
      />
    );
    const textbox = screen.getByRole('textbox');
    textbox && expect(textbox).toHaveValue('');
    textbox && expect(textbox).toBeEnabled();
  });

  it('Should pass when value is changed', async () => {
    const changeMock = jest.fn();
    render(
      <Textarea
        onInputChange={changeMock}
        variant="outline"
        placeholder="test-placeholder"
      />
    );
    const textbox = screen.getByRole('textbox');
    textbox && expect(textbox).toHaveValue('');
    textbox && userEvent.type(textbox, 'test content');
    await waitFor(() => {
      expect(textbox).toHaveValue('test content');
    });
  });
});
