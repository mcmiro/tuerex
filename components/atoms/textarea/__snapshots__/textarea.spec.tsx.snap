// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Textarea component Should pass when name, variant, size and placeholder is additional 1`] = `
<textarea
  class="w-full py-[10px] px-[16px] text-md outline-none border hover:border-2 focus:border-2 disabled:border"
  cols="20"
  id=":r0:"
  placeholder="test"
  rows="4"
/>
`;
